new_bls12_hash_function-milestone_2 (#556)
w3f/Grant-Milestone-Delivery: deliveries/new_bls12_hash_function-milestone_2.md

# Milestone Delivery :mailbox:

**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**  

* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/new_bls12_hash_function.md
* **Milestone Number:** 2

**Context** (optional)

**Deliverables**

| Number | Deliverable | Link | Notes |
| ------------- | ------------- | ------------- |------------- |

> | 0a. | License | Apache License, Version 2.0 (https://www.apache.org/licenses/LICENSE-2.0) and MIT license (https://opensource.org/licenses/MIT) | ... |

Reviewer: Please add a [license file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository#including-an-open-source-license-in-your-repository) to the repository

Author: @alxs, could you give a link to this repository ? I cannot find the necessary page on GitHub.

Reviewer: https://github.com/zhenfeizhang/indifferentiable-hashing

Reviewer 2: added an MIT license: https://github.com/zhenfeizhang/indifferentiable-hashing/blob/master/LICENSE

Reviewer: ```suggestion
| 0a. | License | [LICENSE](https://github.com/zhenfeizhang/indifferentiable-hashing/blob/master/LICENSE) | MIT |
```

> | 0b. | Documentation | published article: https://link.springer.com/article/10.1007/s10623-022-01012-8, free eprint version: https://eprint.iacr.org/2021/301 | ... |

Reviewer: This deliverable refers to inline documentation
```suggestion
| 0b. | Documentation |  | Inline documentation |
```

> | 0c. | Testing Guide | make test | You may need [SageMath](https://www.sagemath.org/) to validate test vectors

Reviewer: ```suggestion
| 0c. | Testing Guide | https://github.com/zhenfeizhang/indifferentiable-hashing#testing | You may need [SageMath](https://www.sagemath.org/) to validate test vectors
```
